contracts/Governance/VestingToken.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.5;
+
+/// Token vesting contract.
+/// Adapted from Melonport AG <team@melonport.com>
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract VestingToken {
+    using SafeMath for uint256;
+
+    ERC20 public token; // Radicle ERC20 contract
+    address public owner; // deployer; can interrupt vesting
+
+    bool public interrupted; // whether vesting is still possible
+    bool public isVestingStarted; // whether vesting period has begun
+    uint256 public totalVestingAmount; // quantity of vested token in total
+    uint256 public vestingStartTime; // timestamp when vesting is set
+    uint256 public vestingPeriod; // total vesting period in seconds
+    address public beneficiary; // address of the beneficiary
+    uint256 public withdrawn; // quantity of token withdrawn so far
+
+    modifier notInterrupted() {
+        require(!interrupted, "The contract has been interrupted");
+        _;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only owner can do this");
+        _;
+    }
+
+    modifier onlyBeneficiary() {
+        require(msg.sender == beneficiary, "Only beneficiary can do this");
+        _;
+    }
+
+    modifier vestingStarted() {
+        require(isVestingStarted, "Vesting must be started");
+        _;
+    }
+
+    /// @param _token Address of token being vested
+    constructor(address _token, address _owner) {
+        token = ERC20(_token);
+        owner = _owner;
+    }
+
+    /// @notice Returns the token amount that is currently withdrawable
+    /// @return withdrawable Quantity of withdrawable Radicle asset
+    function withdrawableBalance() public view returns (uint256 withdrawable) {
+        uint256 timePassed = block.timestamp.sub(vestingStartTime);
+
+        if (timePassed < vestingPeriod) {
+            uint256 vested = totalVestingAmount.mul(timePassed) / vestingPeriod;
+            withdrawable = vested.sub(withdrawn);
+        } else {
+            withdrawable = totalVestingAmount.sub(withdrawn);
+        }
+    }
+
+    /// @notice Grant vesting tokens to the beneficiary.
+    ///
+    /// @param ofBeneficiary Address of beneficiary
+    /// @param ofTokenAmount Amount of tokens
+    /// @param ofVestingPeriod Vesting period in seconds from vestingStartTime
+    function grantTokens(
+        address ofBeneficiary,
+        uint256 ofTokenAmount,
+        uint256 ofVestingPeriod
+    ) external onlyOwner notInterrupted {
```

Comment:
I think that moving these parameters to the constructor would simplify logic and reduce storage access. The vested contract has funds granted only once in its lifetime anyway and it doesn't make sense to have it constructed and not granted.

Reply:
Good point!

Reply:
73824c9

Reply:
Cool! Now almost all the variables can be `immutable` to cut gas usage!

Reply:
contracts/Governance/VestingToken.sol:55:13: TypeError: Immutable variables cannot be read during contract creation time, which means they cannot be read in the constructor or any function or modifier called from it.
--
  | token.transferFrom(msg.sender, address(this), _amount),

Reply:
03be174

Reply:
https://github.com/radicle-dev/radicle-contracts/pull/84/commits/e604d1207da0998087d5436a92b7b26d5dd1fcf7 :)